### PR TITLE
[POT-2] Display companies functionality

### PR DESCRIPTION
**Pull request by pkondratas**
Created API method through which companies are displayed, additional unit tests.

### CODE REVIEW

File: template/src/main/java/pot/insurance/manager/controller/CompanyRestController.java [hunk 1]
```diff
@@ -18,4 +18,10 @@ public class CompanyRestController {
     public CompanyDTO createCompany(@RequestBody CompanyDTO companyDTO) {
         return companyService.saveCompany(companyDTO);
     }
+
+    @GetMapping
+    @ResponseStatus(HttpStatus.OK)
```
Comment (anton-kolyaev) [inline template/src/main/java/pot/insurance/manager/controller/CompanyRestController.java:23]: NIT: Not necessary. OK status set by default

File: template/src/main/java/pot/insurance/manager/service/CompanyService.java [hunk 2]
```diff
@@ -29,10 +30,16 @@ public CompanyDTO saveCompany(CompanyDTO companyDTO) {
...
         try {
             return companyMapper.companyToCompanyDTO(companyRepository.save(company));
         } catch(DataIntegrityViolationException e) {
             throw new DataValidationException(DataValidation.Status.MALFORMED_DATA);
         }
     }
+
+    public Object getAllCompanies() {
+        List<Company> companyList = companyRepository.findAll();
+        return companyList.stream().map(companyMapper::companyToCompanyDTO).toList();
```
Comment (anton-kolyaev) [inline template/src/main/java/pot/insurance/manager/service/CompanyService.java:43]: NIT: put each mapping on own line - that will help you to debug if smth go wrong

**Review by anton-kolyaev — COMMENTED** [review]
It would be useful to add integration test as well.

File: template/src/test/groovy/controller/CompanyRestControllerSpec.groovy [hunk 3]
```diff
@@ -3,14 +3,15 @@ package controller
...
+import pot.insurance.manager.entity.Company
 import pot.insurance.manager.service.CompanyService
 import spock.lang.Specification
 
 class CompanyRestControllerSpec extends Specification implements TestableTrait {
     def companyService = Mock(CompanyService)
     def controller = new CompanyRestController(companyService)
 
-    def "createCompany test"() {
+    def "test for createCompany method"() {
```
Comment (anton-kolyaev) [inline template/src/test/groovy/controller/CompanyRestControllerSpec.groovy:14]: that is obvious that it is a `test` better to mention what actually it checks and what is expected

File: template/src/main/resources/application.yaml [hunk 4]
```diff
@@ -1,3 +1,4 @@
+debug: true
```
Comment (anton-kolyaev) [inline template/src/main/resources/application.yaml:1]: should it be here? Probably you need in it in separate `application-local.yaml` file

**Review by anton-kolyaev — DISMISSED** [review]

**Review by NorthDragonDuck — APPROVED** [review]

**Comment by pkondratas** [comment]
Renamed tests, and created application-local.yaml for debug log level.

File: template/src/main/resources/application.yaml [hunk 5]
```diff
@@ -1,4 +1,6 @@
 spring:
+    profiles:
```
Comment (anton-kolyaev) [inline template/src/main/resources/application.yaml:2]: leave default profile for prod (just remove all from here). To run your spring use
`./gradlew bootRun --args='--spring.profiles.active=local'`
or set profile in IntellijIDEA run config

File: template/src/test/groovy/integration/CompanyIntegrationSpec.groovy [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+                .andReturn()
+                .response
+
+        then:
+        result.contentAsString.equals(json)
+        result.getStatus() == 201
+
+        where:
+               id         | code |        name       |      site     |     email
+        UUID.randomUUID() | "US" | "Example company" | "example.com" | "email@gmail.com"
```
Comment (anton-kolyaev) [inline template/src/test/groovy/integration/CompanyIntegrationSpec.groovy:47]: you can add `expected` column to this table (i.e. status code) and then you'll be able test not the only succeed requests

**Review by NorthDragonDuck — APPROVED** [review]

**Pull request merged**